### ISSUE TITLE
Chokes when accessing dotted properties

### ISSUE THREAD

**Issue by dvv** [node-red/node-red#899]
Feeding the following node

```js
[
  {
    "id": "c9973eaf.460b7",
    "type": "switch",
    "z": "5061b0ff.2966f",
    "name": "",
    "property": "state[\"switch.room.wrkr\"]",
    "propertyType": "msg",
    "rules": [
      {
        "t": "eq",
        "v": "0",
        "vt": "str"
      }
    ],
    "checkall": "true",
    "outputs": 1,
    "x": 830,
    "y": 640,
    "wires": [
      [
        "e9e3c466.409d58"
      ]
    ]
  }
]
```

with a message
```js
{
  topic: "foo",
  payload: "bar",
  state: {
    "switch.room.wrkr": 1
  }
}
```

gives `TypeError: Cannot read property 'room' of undefined`.

Is it intended or a bug?

TIA,
--Vladimir

**Comment by knolleary** [comment]
We only support the dot-syntax for specifying properties, which means we don't support property names that have a dot in.
This is a current limitation, but one we need to fix.
I'll see if we can get it resolved for the imminent 0.14.

**Comment by dvv** [comment]
I see. So far I just used a "function" node for filtering based on dotted props.
Thank you!

**Comment by knolleary** [comment]
Fixed in the 0.14.0 branch thanks to 762eb07dd4b95a555d0abf4933d042e5980fd426

Supports expressions such as:

     msg.a.b
     msg.a['b']
     msg.a[1]
     msg.a[1][2].b["foo.bar[2]"]

etc
